internal/provider/cluster_data_source.go

```diff
@@ -362,39 +365,51 @@ func populateClusterDataSource(
 
 	// Populate LoginNodes
 	if loginNodes, ok := cluster.GetLoginNodesOk(); ok {
-		loginNodesMap, err := loginNodes.ToMap()
-		if err != nil {
-			resp.Diagnostics.AddError(
-				"Error occured while retrieving cluster list",
-				fmt.Sprintf("Error: %v", err),
-			)
-		}
-		delete(loginNodesMap, "healthyNodes")
-		delete(loginNodesMap, "unhealthyNodes")
-
-		loginNodesObjectMap, diags := types.MapValueFrom(
-			ctx,
-			types.StringType,
-			loginNodesMap,
-		)
-		loginNodesObjectMapElements := loginNodesObjectMap.Elements()
-
-		loginNodesObjectMapElements["healthyNodes"] = types.NumberValue(
-			big.NewFloat(float64(loginNodes.GetHealthyNodes())),
-		)
-		loginNodesObjectMapElements["unhealthyNodes"] = types.NumberValue(
-			big.NewFloat(float64(loginNodes.GetUnhealthyNodes())),
-		)
-
-		resp.Diagnostics.Append(diags...)
-		loginNodesObject, diags := types.ObjectValue(
-			loginNodesObjectTypes,
-			loginNodesObjectMapElements,
-		)
-		resp.Diagnostics.Append(diags...)
-		tfClusterMapElements["loginNodes"] = loginNodesObject
+	    loginNodesList := make([]types.Object, 0)
+	    for _, loginNode := range loginNodes{
+	        loginNodeMap, err := loginNode.ToMap()
+            if err != nil {
+                resp.Diagnostics.AddError(
+                    "Error occured while retrieving cluster list",
+                    fmt.Sprintf("Error: %v", err),
+                )
+            }
+            delete(loginNodeMap, "healthyNodes")
+            delete(loginNodeMap, "unhealthyNodes")
+
+            loginNodeObjectMap, diags := types.MapValueFrom(
+                ctx,
+                types.StringType,
+                loginNodeMap,
+            )
+            loginNodeObjectMapElements := loginNodeObjectMap.Elements()
+
+            loginNodeObjectMapElements["healthyNodes"] = types.NumberValue(
+                big.NewFloat(float64(loginNode.GetHealthyNodes())),
+            )
+            loginNodeObjectMapElements["unhealthyNodes"] = types.NumberValue(
+                big.NewFloat(float64(loginNode.GetUnhealthyNodes())),
```

Comment:
why float? i think this could be an integer

Reply:
The previous code used Float so I just did the same thing

Reply:
ok, let's fix this in a follow up Pr then